packages/node_modules/@node-red/nodes/core/common/lib/debug/debug-utils.js

```diff
@@ -536,10 +536,20 @@ RED.debug = (function() {
                 e.stopPropagation();
                 showMessageMenu(filterMessage,debugMessage,sourceNode&&sourceNode.id);
             });
-            $('<span class="red-ui-debug-msg-topic">'+
-                (o.topic?topic+' : ':'')+
-                (o.property?'msg.'+property:'msg')+" : "+format+
-                '</span>').appendTo(metaRow);
+            const msgInfos = `${o.topic ? " : " : ""}msg${o.property ? "." + property : ""} : ${format}`
+            const topicTruncated = RED.utils.truncateString((o.topic ? topic : ""), 120 - msgInfos.length);
+            const topicElem = $('<span class="red-ui-debug-msg-topic">' + topicTruncated + msgInfos + '</span>').appendTo(metaRow);
+            RED.popover.create({
+                content: topic,
+                delay: { show: 750, hide: 50 },
+                direction: "bottom",
+                interactive: true,
+                maxWidth: 300,
+                target: topicElem,
+                trigger: "hover",
+                tooltip: true,
+                size: "small"
+            });
```

Comment:
```suggestion
            if (topic !== topicTruncated) {
                RED.popover.create({
                    content: topic,
                    delay: { show: 750, hide: 50 },
                    direction: "bottom",
                    interactive: true,
                    maxWidth: 300,
                    target: topicElem,
                    trigger: "hover",
                    tooltip: true,
                    size: "small"
                });
            }
```

Suggest only adding tooltip if there's a truncated topic; otherwise its additional overhead on each debug message that will mostly get unused.